src/psyclone/tests/domain/common/algorithm/alg_psyir_test.py

```diff
@@ -526,14 +563,105 @@ def test_aic_lowertolanguagelevel_multi_invoke():
 
     assert len(psyir.walk(AlgorithmInvokeCall)) == 0
     assert len(psyir.walk(KernelFunctor)) == 0
-
     call1 = psyir.children[0][0]
+    with pytest.raises(KeyError) as info:
+        call1.scope.symbol_table.lookup("invoke")
+    assert "Could not find 'invoke' in the Symbol Table." in str(info.value)
+    call2 = psyir.children[0][1]
+    with pytest.raises(KeyError) as info:
+        call2.scope.symbol_table.lookup("invoke")
+    assert "Could not find 'invoke' in the Symbol Table." in str(info.value)
+
     assert call1.routine.name == "invoke_0_kern1_1"
     assert call1.routine.interface.container_symbol.name == "psy_alg1_1"
 
-    call2 = psyir.children[0][1]
     assert call2.routine.name == "invoke_1_kern2"
-    assert call1.routine.interface.container_symbol.name == "psy_alg1_1"
+    assert call2.routine.interface.container_symbol.name == "psy_alg1_1"
+
+
+def test_aic_lowertolanguagelevel_invoke_symbols():
+    '''Check that the lower_to_language_level method removes the
+    appropriate invoke symbols when these symbols are within symbol
+    tables that are connected to different parts of the PSyIR tree,
+    including where the nodes are ancestors of each other.
+
+    '''
+    code = (
+        "module mod1\n"
+        "contains\n"
+        "subroutine alg1()\n"
+        "  use kern_mod, only : kern1\n"
+        "  use field_mod, only : field_type\n"
+        "  type(field_type) :: field\n"
+        "  call invoke(kern1(field))\n"
+        "end subroutine alg1\n"
+        "subroutine alg2()\n"
+        "  use kern_mod, only : kern2\n"
+        "  use field_mod, only : field_type\n"
+        "  type(field_type) :: field\n"
+        "  call invoke(kern2(field))\n"
+        "end subroutine alg2\n"
+        "end module\n")
+
+    psyir = create_alg_psyir(code)
+    # Add an invoke symbol to the module (Container node)
+    container = psyir.children[0]
+    assert isinstance(container, Container)
+    container.symbol_table.add(RoutineSymbol("invoke"))
+
+    invoke1 = container.children[0][0]
+    assert isinstance(invoke1, AlgorithmInvokeCall)
+    invoke2 = container.children[1][0]
+    assert isinstance(invoke2, AlgorithmInvokeCall)
+
+    assert invoke1.scope.symbol_table.lookup(
+        "invoke", scope_limit=invoke1.scope)
+    assert invoke2.scope.symbol_table.lookup(
+        "invoke", scope_limit=invoke2.scope)
+
+    psyir.lower_to_language_level()
+
+    call1 = container.children[0][0]
+    call2 = container.children[1][0]
+    with pytest.raises(KeyError) as info:
+        assert call1.scope.symbol_table.lookup(
```

Comment:
Again, could be simplified by just checking that the symbol isn't inside the specific table.

Reply:
Agreed. Updated.